README.md

```diff
@@ -1,6 +1,233 @@
-create and add your own custom copilots to enhance the capabilities of Terminal
-Pro.
+Welcome to the example repository for integrating a custom copilot into the OpenBB Terminal Pro!
+
+This guide provides everything you need to create and add your own custom copilots, enhancing the capabilities of Terminal Pro in a way that fits your unique requirements.
+
+## Overview
+
+To integrate a custom copilot, you'll need to create a backend that interacts with Terminal Pro. Terminal Pro will send requests to your copilot, which will respond accordingly via Server-Sent Events ([SSEs](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events)). This ensures real-time responses and seamless communication between Terminal Pro and your custom copilot.
+
+This guide will cover:
+
+- Understanding the message and context structure
+- How to handle function calls and interact with widgets
+- Explanation of the copilot configuration (copilots.json)
+- Examples of common use cases and scenarios
+
+## Requirements for Compatibility
+
+Custom copilots are expected to follow the defined schema for queries, SSE responses, and function calling. These requirements ensure seamless integration with OpenBB Terminal Pro.
+
+## How Terminal Pro submits requests to your Copilot
+
+Terminal Pro will submit POST requests to your custom copilot's `/v1/query` endpoint. These requests will contain information such as the user's message  (`messages`), any additional `context`, `widgets` data, user-uploaded files (`user_files`), relevant URLs (`urls`), and more. Your copilot must respond via SSEs for real-time interaction.
+
+### API Request Schema Example
+
+```json
+{
+  "messages": [
+    {
+      "role": "human",
+      "content": "Hi there. Who are you?"
+    }
+  ],
+  "context": null,
+  "widgets": [],
+  "urls": []
+}
+```
+
+- messages: The chat history, including both user and assistant messages.
+- context: Structured context to include additional information like personal data or metadata.
+- widgets: A list of widgets currently in use by the user for function calling.
+- user_files: UUIDs of uploaded files.
+- urls: A list of URLs for web search functionality.
+
+**Note:** Terminal Pro makes requests to this endpoint each time the user interacts with the copilot.
+
+## How the Copilot Responds
+
+Your custom copilot must respond to Terminal Pro's request using Server-Sent Events (SSEs).
+
+Here's a breakdown of how your copilot can respond:
+
+Sending a `copilotMessageChunk` Event
+
+The copilotMessageChunk event is used to stream back generated tokens (partial responses) to Terminal Pro. These responses are streamed as they are generated, providing a seamless experience to the use
+
+``` json
+{
+  "event": "copilotMessageChunk",
+  "data": {
+    "delta": "<token>"
+  }
+}
+```
+
+These delta values are streamed by Terminal Pro to form the final message.
+
+## Adding Function Calling to your Custom Copilot
```

Comment:
You'll need to talk about the various function calls that Terminal currently supports (for now, just the one -- `get_widget_data`).